src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/VaultDepositWithdrawForm.kt

```diff
@@ -0,0 +1,274 @@
+package exchange.dydx.abacus.functional.vault
+
+import kotlinx.serialization.Serializable
+import kotlin.js.JsExport
+
+@JsExport
+@Serializable
+data class VaultFormData(
+    val action: VaultFormAction,
+    val amount: Double?,
+    val acknowledgedSlippage: Boolean,
+    val inConfirmationStep: Boolean,
+)
+
+@JsExport
+@Serializable
+enum class VaultFormAction {
+    DEPOSIT,
+    WITHDRAW,
+}
+
+@JsExport
+@Serializable
+data class VaultFormAccountData(
+    val marginUsage: Double?,
+    val freeCollateral: Double?
+)
+
+@JsExport
+@Serializable
+data class VaultDepositWithdrawSlippageResponse(
+    val shares: Double,
+    val expectedAmount: Double,
+)
+
+@JsExport
+@Serializable
+data class VaultFormValidationError(
+    val severity: VaultFormValidationErrorSeverity,
+    val type: VaultFormValidationErrorType
+)
+
+@JsExport
+@Serializable
+enum class VaultFormValidationErrorSeverity {
+    WARNING,
+    ERROR
+}
```

Comment:
I don't think we need to add ErrorSeverity for every form that we add. There is an existing `ErrorType` from `output/input/ValidationError` that includes error, warning, required.